compiler+runtime/src/cpp/jank/runtime/core.cpp

```diff
@@ -245,7 +245,7 @@ namespace jank::runtime
     }
     else if(!rhs)
     {
-      return !lhs;
+      return false;
```

Comment:
Why change this? Right now, it would treat two null object ptrs as equal. This would also carry over to us removing `nullptr` support for boxes, such that two nils would be equal.

Reply:
Thanks, I missed that box overloaded `!`.

Reply:
Actually, could you look at this again? At this point we already know `!lhs` is `false`, so I think returning either is equivalent.

Would this be another way of writing the test?

```c++
if(!lhs != !rhs) return false
```

Reply:
Added assertions to point out my reasoning.